### ISSUE TITLE
"LLVM ERROR: Broken module found, compilation aborted" with closures

### ISSUE THREAD

**Issue by Kimundi** [rust-lang/rust#23960]
```rust
#![feature(unboxed_closures)]

fn run_double_ended_test<T, TF, F, IF, E>(_: TF, _: F, _: Vec<E>)
    where TF: Fn() -> T,
          F: Fn<T, Output=IF>,
          IF: Iterator<Item=E>,
{}

fn double_ended_split_char2() {
    run_double_ended_test(|| (|| "foo.bar.baz", || '.'),
                          |s, p| (str::split)(s(), p()),
                          vec!["foo", "bar", "baz"]);
}

fn main() { double_ended_split_char2(); }
```
```


Global is external, but doesn't have external or weak linkage!
void (%str_slice*, %closure.1*)* @_ZN24double_ended_split_char212closure.1329E
invalid linkage type for function declaration
void (%str_slice*, %closure.1*)* @_ZN24double_ended_split_char212closure.1329E
Global is external, but doesn't have external or weak linkage!
i32 (%closure.2*)* @_ZN24double_ended_split_char212closure.1331E
invalid linkage type for function declaration
i32 (%closure.2*)* @_ZN24double_ended_split_char212closure.1331E
LLVM ERROR: Broken module found, compilation aborted!
playpen: application terminated with error code 1
Program ended.


```

**Comment by sanxiyn** [comment]
This seems fixed, presumably by #26079.

**Comment by alexcrichton** [comment]
Yay!
